File: GraphsJava/TopoSort.java
class Solution
{
    //Function to return list containing vertices in Topological order. 
    static int[] topoSort(int N, ArrayList<ArrayList<Integer>> adj) 
    {
        // add your code here
        int topo[] = new int[N]; 
        int indegree[] = new int[N]; 
        for(int i = 0;i<N;i++) {
            for(Integer it: adj.get(i)) {
                indegree[it]++; 
            }
        }
        
        Queue<Integer> q = new LinkedList<Integer>(); 
        for(int i = 0;i<N;i++) {
            if(indegree[i] == 0) {
                q.add(i); 
            }
        }
        int cnt = 0;
        while(!q.isEmpty()) {
            Integer node = q.poll(); 
            topo[cnt] = node;
            cnt++; 
            
            for(Integer it: adj.get(node)) {
                indegree[it]--; 
                if(indegree[it] == 0) {
                    q.add(it); 
                }
            }
        }
        // if(cnt == N) return false; 
        // return true; 
        return topo;
    }
}

File: C++/GraphsCPP/topoSort.cpp
Topological Sort/Kahn Algorithm - GFG

class Solution
{
	public:
	//Function to return list containing vertices in Topological order. 
	vector<int> topoSort(int V, vector<int> adj[]) 
	{
	    // code here
	    vector<int> res;
	    vector<int> in(V, 0);
	    
	    queue<int> q;
	    
	    for(int i = 0; i < V; i++){
	        for(auto x: adj[i]){
	            in[x]++;
	        }   
	    }
	    
	    for(int i = 0; i < V; i++){
	        if(in[i]==0)
	            q.push(i);
	    }
	    
	    while(!q.empty()){
	        int temp = q.front();
	        res.push_back(temp);
	        q.pop();
	        
	        for(auto x: adj[temp]){
	            in[x]--;
	            if(in[x]==0){
	                q.push(x);
	            }
	        }
	    }
	    
	    return res;
	}
};
